### ISSUE TITLE
ps_emailalerts: As a merchant be informed about customers subscribing to product availability notifications

### ISSUE THREAD

**Issue by lmeyer1** [PrestaShop/PrestaShop#28233]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

If a customer subscribes to a notification for a product out of stock, the merchant should be aware. In fact, the merchant may have to take action.

### Describe the solution you'd like

There are two approches, that could be complementary.
- In the BO, the merchant could choose to receive a mail when a customer subscribes to a notification
- There should be an entry `Out of stock notifications` in the menu `Catalog` that displays all active notifications.

### Alternatives you've considered

We could create a module that does just that: list all active notifications.

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by marwachelly** [comment]
Hello @lmeyer1 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon.

We're waiting for your PR :rocket: 

Thanks ,

**Comment by MatShir** [comment]
We might need to list the notification by product 🤔 
I'm wondering if this feature should go into Catalog > Monitoring. What do you think? 
Otherwise, nice feature :)

**Comment by lmeyer1** [comment]
@MatShir 
> I'm wondering if this feature should go into Catalog > Monitoring. What do you think?

I'm not convinced. In _Monitoring_, all lists are related strictly to products and categories.
These notifications are stronger related to carts and orders.
Also, since it is a module, it would be easier, if this list is separate menu entry, as it can be configured for module admin controllers.

**Comment by MatShir** [comment]
You are probably right ! 
So list of product with their name, refs, number of user waiting. When you click on a product, you are redirected to an email list (linked to the user account or not). 

If we collect the email, we need to check the RGPD module

**Comment by lmeyer1** [comment]
@MatShir Yes that is it. The integration with RGPD module is already in place.
I will come back to this issue later.

**Comment by SharakPL** [comment]
It's still relevant

**Comment by boo-code** [comment]
The data for both halves already exists: a subscription is a `MailAlert` row, created in `controllers/front/actions.php` right after the email is validated, and the module installs its own table for them.

The first half also has a pattern to follow rather than a design to invent. The module already carries four merchant notifications, each a setting plus an optional recipient list plus one mail template:

| setting | recipients | template |
| ------- | ---------- | -------- |
| `MA_MERCHANT_ORDER` | `MA_MERCHANT_ORDER_EMAILS` | `new_order` |
| `MA_MERCHANT_OOS` | `MA_MERCHANT_OOS_EMAILS` | `productoutofstock` |
| `MA_MERCHANT_COVERAGE` | | `productcoverage` |
| `MA_RETURN_SLIP` | `MA_RETURN_SLIP_EMAILS` | `return_slip` |

So "tell the merchant someone subscribed" is one more row in that table: a setting pair, one template pair, and a send where the `MailAlert` is added. The module ships `mails/en` and `mails/fr` only, so that is two template files per format rather than one per shipped language.

The second half is the one that needs the decision this thread is stuck on, and there is a constraint worth adding to it: the module has no admin controller at all today, only `controllers/front`. A list of subscriptions means introducing one, wherever it ends up living, which is a larger step than the notification and is probably why the two halves are worth separating rather than specifying together.

One caveat on ordering, since it affects whoever picks this up: `ps_emailalerts.php` currently has open pull requests against it, so a change adding a fifth merchant notification will want to land after those rather than beside them.
